### PR TITLE
Postfix sasl jails

### PR DESCRIPTION
**Pull request by stephdl**
Enable sasl jails for smtpd

jail for bad smtpd login (postfix-sasl)
jail for abuse usage of smtpd (33*MaxRetry, 99 emails sent in 900 seconds is the default)

the IP of the abuser is banned

https://github.com/NethServer/dev/issues/5618

### CODE REVIEW

File: root/etc/fail2ban/filter.d/postfix-sasl-abuse.conf [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+[INCLUDES]
+
+before = common.conf
+
+[Definition]
+
+_daemon = postfix(-\w+)?/(?:submission/|smtps/)?smtp[ds]
+
+# This regex is the full matched line but it could probably miss something, I prefer to use the last shortcut regex
+#failregex = ^%(__prefix_line)s\w+: client=(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])\[<HOST>\], sasl_method=(LOGIN|PLAIN|(?:CRAM|DIGEST)-MD5), sasl_username=[^@]+@\S+$
```
Comment (DavidePrincipi) [inline root/etc/fail2ban/filter.d/postfix-sasl-abuse.conf]: I'd remove the commented line, and add it to the docs if you feel it can help

**Comment by stephdl** [comment]
No issue for now, just a first try to check if you think it valuable :)

File: root/etc/fail2ban/filter.d/postfix-sasl-abuse.conf (hunk 1 above)
Comment (stephdl) [reply root/etc/fail2ban/filter.d/postfix-sasl-abuse.conf]: in fact either we need to match `unknow` or a full long hostname like `abordeaux-656-1-342-200.w90-55.abo.wanadoo.fr` in short for reference

```
Oct 27 17:15:39 prometheus postfix/smtpd[27043]: DD2A91901DE14: client=unknown[80.12.58.22], sasl_method=PLAIN, sasl_username=stephane@domain.com
Oct 27 18:36:11 prometheus postfix/smtpd[4583]: 6BDF21901DE17: client=abordeaux-656-1-342-200.w90-55.abo.wanadoo.fr[90.55.182.200], sasl_method=PLAIN, sasl_username=stephane@domain.com
```

**Comment by gsanchietti** [comment]
Before merge, wait for release of #43 and #42

**Comment by filippocarletti** [comment]
I'd prefer to keep the current policy: jails enabled by default with "good" ban settings.
So the question is: are the proposed settings good enough?
We can only discover by trial and error. I think that fail2ban users will keep an eye on their logs.
Maybe a wider audience (and a word of warning to users) could help us to release better defaults.

**Comment by DavidePrincipi** [comment]
Isn't it disabled by default?

**Comment by stephdl** [comment]
The jail is enabled per default, the limit is 99 email in 900 seconds, and if the server send itself the email by the localhost for a mailling list, the localhost is whitelisted.

We have also a specific maxRetry prop for this jail, this could be changed easily without custom template.

The proposal the @filippocarletti makes sense, does the value 99 is good enough

**Comment by stephdl** [comment]
we could also add a specific ignoreIP here only for this jail, like this remote IP will be enabled

**Pull request merged**